packages/http-cors/index.js

```diff
@@ -22,123 +69,53 @@ const defaults = {
   methods: undefined,
   origin: '*',
   origins: [],
+  enablePreflightReply: true,
   exposeHeaders: undefined,
   maxAge: undefined,
   requestHeaders: undefined,
   requestMethods: undefined,
   cacheControl: undefined,
   vary: undefined
-}
-
+};
 const httpCorsMiddleware = (opts = {}) => {
-  const options = { ...defaults, ...opts }
-
-  const httpCorsMiddlewareAfter = async (request) => {
-    normalizeHttpResponse(request)
-
-    const { headers } = request.response
-    const existingHeaders = Object.keys(headers)
-
-    // Check if already setup the header Access-Control-Allow-Credentials
-    if (existingHeaders.includes('Access-Control-Allow-Credentials')) {
-      options.credentials =
-        headers['Access-Control-Allow-Credentials'] === 'true'
+  const options = {
+    ...defaults,
+    ...opts
+  };
+  const httpCorsMiddlewareBefore = async (request) => {
+    if (options.enablePreflightReply) {
+      if (request.event.httpMethod === "OPTIONS") {
```

Comment:
You should use the snippet below to ensure it works for all http event versions.
```
const httpMethod = getVersionHttpMethod[request.event.version ?? '1.0']?.(
      request.event
    )
    ```